deps/media-playback/media-playback/media.c

```diff
@@ -653,93 +637,349 @@ static bool init_avformat(mp_media_t *m)
 	return true;
 }
 
-static void reset_ts(mp_media_t *m)
+/*Function adjusts timestamp after pausing to avoid fast forward effect.*/
+static inline void reset_ts(mp_media_t *m)
 {
 	m->base_ts += mp_media_get_base_pts(m);
 	m->play_sys_ts = (int64_t)os_gettime_ns();
 	m->start_ts = m->next_pts_ns = mp_media_get_next_min_pts(m);
 	m->next_ns = 0;
 }
 
-static inline bool mp_media_thread(mp_media_t *m)
+/*Renders next frame*/
+static inline void mp_render_frame(mp_media_t *m)
 {
-	os_set_thread_name("mp_media_thread");
+	if (m->has_video)
+		mp_media_next_video(m, false);
+	if (m->has_audio)
+		mp_media_next_audio(m);
 
-	if (!init_avformat(m)) {
-		return false;
+	if (!mp_media_prepare_frames(m))
+		return;
+	if (!mp_media_eof(m))
+		mp_media_calc_next_ns(m);
+}
+
+/*Execute when media enters playing state*/
+static inline bool mp_media_playing(mp_media_t *m)
+{
+	m->mp_state = mstate_playing;
+	return true;
+}
+
+/*Execute when media enters pausing state*/
+static inline bool mp_media_pausing(mp_media_t *m)
+{
+	if (m->mp_state != mstate_stopping && m->mp_state != mstate_stopped)
+		m->mp_state = mstate_paused;
+	return true;
+}
+
+/*Execute when media moves from paused to playing.*/
+static inline bool mp_media_resume(mp_media_t *m)
+{
+	if (m->mp_state == mstate_paused) {
+		m->mp_state = mstate_resume;
+
+		mp_media_push_message(m, mstate_playing);
 	}
-	if (!mp_media_reset(m)) {
+	return true;
+}
+
+static inline bool mp_media_seeking(mp_media_t *m)
+{
+	enum mp_media_state oldstate = m->mp_state;
+
+	m->mp_state = mstate_seeking;
+	int64_t seek_pos;
+
+	seek_pos = m->seek_pos;
+	m->seek_next_ts = true;
+
+	seek_to(m, seek_pos);
+
+	if (!m->eof) {
+		switch (oldstate) {
+		case mstate_paused:
+			mp_media_push_message(m, mstate_paused);
+			break;
+		default:
+			mp_media_push_message(m, mstate_playing);
+			break;
+		}
+	}
+	return true;
+}
+
+/*Executes when stop event occurs. One time action,
+  then enter the stopped state. */
+static inline bool mp_media_stopping(mp_media_t *m)
+{
+	m->mp_state = mstate_stopping;
+
+	mp_media_push_message(m, mstate_stopped);
+
+	return true;
+}
+
+/*Occurs after stopping but before another state*/
+static inline bool mp_media_stopped(mp_media_t *m)
+{
+	m->mp_state = mstate_stopped;
+
+	if (m->stop_cb)
+		m->stop_cb(m->opaque);
+
+	return true;
+}
+
+/*Flip the looping flag*/
+static inline bool mp_media_setloop(mp_media_t *m)
+{
+	enum mp_media_state oldstate = m->mp_state;
+
```

Comment:
Remove the extra blank newlines in this function.  A single empty newline before the return statement should suffice.